public/js/fxa-analytics.js

```diff
@@ -49,15 +49,38 @@ async function sendPing(el, eventAction, eventLabel = null) {
   }
 }
 
+
 function getFxaUtms(url) {
-  const utmSource = encodeURIComponent(document.body.dataset.serverUrl.replace(/(^\w+:|^)\/\//g, ""));
-  url.searchParams.append("utm_source", utmSource);
-  url.searchParams.append("utm_campaign", document.body.dataset.utmCampaign);
+  const bodyDataset = document.body.dataset;
+  getUTMNames().forEach(param => {
+    if (bodyDataset[param]) {
+      url.searchParams.append(param, encodeURIComponent(bodyDataset[param]));
+    }
+  });
   url.searchParams.append("form_type", "email");
   return url;
 }
 
+function saveReferringPageData(utmParams) {
+  const bodyDataset = document.body.dataset;
+  getUTMNames().forEach(param => {
+    if (utmParams.has(param)) {
+      const cleanParam = utmParams.get(param);
+      bodyDataset[param] = cleanParam.replace(/[&<>"',.`=:/]/g, "");
+    }
+  });
+}
+
+function getUTMNames() {
+  return ["utm_source", "utm_medium", "utm_campaign", "utm_term", "utm_content"];
+}
+
 (() => {
+  const win = window;
+  if (win.location.search.includes("utm_") && win.history.replaceState) {
+    saveReferringPageData(new URL(win.location).searchParams);
+    win.history.replaceState({}, "", win.location.toString().replace(/[?&]utm_.*/g, ""));
```

Comment:
This may also be a bit greedy.

I tried in regexr.com with this[1] made up URL, and it takes everything after `utm_` to the end of the line, even the &age=999&foo=bar. Not sure if we ever expect anything else on the querystring apart from UTMs, but just throwing that out there.

```
[1] https://monitor.firefox.com/scanner/?name=peter&utm_source=ok&age=999&foo=bar
```

Reply:
So I think the UTMs are technically supposed to be at the end of the URL, in which case clipping everything after the `?utm_` is what we want and I don't think we're currently wanting any values that could feasibly come after the UTMs to stick around in the url bar after the UTMs have been clipped out in any case.